parsers/ZA.py

```diff
@@ -708,6 +708,27 @@ def get_power_bi_values():
     return cleaned_power_bi_values
 
 
+def valid_substitution(original, subs_key):
```

Comment:
```suggestion
def valid_substitution(original: str, subs_key: str):
```
Adding type hints here will make sure the `original` (csv) and `sub_key` is always a string and it will have cascading effects in the function.